### PR TITLE
BNH-176: Ensuring server-side validation and buttons on tenant preference form page function correctly

### PR DESCRIPTION
**Pull request by SleepyUser**
-Users can no longer submit their preferences without selecting at least one of Single, Couple, Family
image

-Users no longer have their form page reset to blank if the model is invalid 
-The Single, Couple, Family buttons now style correctly, appearing filled in with colour if the underlying value is true.
image

### CODE REVIEW

File: web/Views/Property/PropertyInputForm/TenantPreferences.cshtml [hunk 1]
```diff
@@ -7,15 +7,15 @@
...
         <div class="col-auto d-grid">
-            @Html.CheckBox("AcceptsSingleTenant", Model.AcceptsSingleTenant ?? false, new { @class = "btn-check" })
-            <label asp-for="AcceptsSingleTenant" class="btn btn-outline-blue shadow-none pill-check">Single tenant</label>
+            <input type="checkbox" asp-for="AcceptsSingleTenant" class="btn-check"/>
+            <label asp-for="AcceptsSingleTenant" id="AcceptsSingleTenant" class="btn btn-outline-blue shadow-none pill-check">Single tenant</label>
         </div>
         <div class="col-auto d-grid">
-            @Html.CheckBox("AcceptsCouple", Model.AcceptsCouple ?? false, new { @class = "btn-check" })
+            <input type="checkbox" asp-for="AcceptsCouple" class="btn-check"/>
             <label asp-for="AcceptsCouple" class="btn btn-outline-blue shadow-none pill-check">Couple</label>
```
Comment (oskarwilliams378) [inline web/Views/Property/PropertyInputForm/TenantPreferences.cshtml:15]: Should these labels gain IDs too?

File: web/wwwroot/js/site.js [hunk 2]
```diff
@@ -25,6 +25,7 @@ function makeRequiredIfSelectedValueIsTarget(target, id, idToRequire) {
     }
 }
 
+
```
Comment (oskarwilliams378) [inline web/wwwroot/js/site.js:28]: extra new line

File: web/Views/Property/PropertyInputForm/TenantPreferences.cshtml [hunk 3]
```diff
@@ -7,15 +7,15 @@
 
     <div class="row g-3">
         <div class="col-auto d-grid">
-            @Html.CheckBox("AcceptsSingleTenant", Model.AcceptsSingleTenant ?? false, new { @class = "btn-check" })
-            <label asp-for="AcceptsSingleTenant" class="btn btn-outline-blue shadow-none pill-check">Single tenant</label>
+            <input type="checkbox" asp-for="AcceptsSingleTenant" class="btn-check"/>
```
Comment (oskarwilliams378) [inline web/Views/Property/PropertyInputForm/TenantPreferences.cshtml:10]: Happy to use tags, we can also use Html.CheckBoxFor if you prefer the other method (I am very happy either way :) )

**Review by oskarwilliams378 — CHANGES_REQUESTED** [review]

File: web/Views/Property/PropertyInputForm/TenantPreferences.cshtml (hunk 1 above)
Comment (SleepyUser) [reply web/Views/Property/PropertyInputForm/TenantPreferences.cshtml:15]: The labels are the buttons, so the asp-for is intended to give them the id and name to allow the user to click it and change the underlying boolean. The input checkbox is hidden, but linked to the same thing, allowing the CSS to look for "input:checked ~ label" and show the label button as pressed.

Comment (oskarwilliams378) [reply web/Views/Property/PropertyInputForm/TenantPreferences.cshtml:15]: In which case should we remove the ID from the button above?

Comment (SleepyUser) [reply web/Views/Property/PropertyInputForm/TenantPreferences.cshtml:15]: Yes! :) fixed

**Review by oskarwilliams378 — APPROVED** [review]

**Pull request merged**